src/bin/pg_autoctl/pidfile.h

```diff
@@ -12,12 +12,55 @@
 #include <inttypes.h>
 #include <signal.h>
 
+#include "postgres_fe.h"
+#include "pqexpbuffer.h"
+
 #include "keeper.h"
 #include "keeper_config.h"
 #include "monitor.h"
 #include "monitor_config.h"
 
+/*
+ * As of pg_autoctl 1.4, the contents of the pidfile is:
+ *
+ * line #
+ *		1	supervisor PID
+ *		2	data directory path
+ *		3	version number (PG_AUTOCTL_VERSION)
+ *		4	extension version number (PG_AUTOCTL_EXTENSION_VERSION)
+ *		5	shared semaphore id (used to serialize log writes)
+ *		6	first supervised service pid line
+ *		7	second supervised service pid line
+ *    ...
+ *
+ * The supervised service lines are added later, not the first time we create
+ * the pidfile. Each service line contains 2 bits of information, separated
+ * with spaces:
+ *
+ *   pid service-name
+ *
+ * Each service creates its own pidfile with its own version number. At
+ * pg_autoctl upgrade time, we might have a supervisor process that's running
+ * with a different version than one of the restarted pg_autoctl services.
+ */
+#define PIDFILE_LINE_PID 1
```

Comment:
Why are we using 1 based indexing instead of 0 based indexing here? It seems that it only makes the rest of the code more complex, by needing two different counters.

Reply:
Postgres itself does it that way in `src/include/utils/pidfile.h` so I wanted to do it the same way as Postgres. Then they have the same off-by-one when reading the file. If that technique is good enough for them, it's good enough 
for us, that's our motto, so I though I would just “do as Postgres do” here.


From Postgres `src/bin/pg_ctl/pg_ctl.c` we can read:

``` c
			pmpid = atol(optlines[LOCK_FILE_LINE_PID - 1]);
			pmstart = atol(optlines[LOCK_FILE_LINE_START_TIME - 1]);
```